bsp_diff/caas/device/intel/civ/host/vm-manager/0046-Multi-Android-VM-support.patch

```diff
@@ -0,0 +1,407 @@
+ }
+ 
++function create_vm_dir() {
++        if [ -e $CIV_DIR/$VM_NAME ]
++        then
++                echo "Folder with name $VM_NAME already present. Delete and create new folder? Please enter yes/no"
++		read input
++		if [ $input = "yes" ]; then
++	                rm -rf $CIV_DIR/$VM_NAME
++	        else
++	                exit
++	        fi
++        fi
++        echo "Dir: $CIV_WORK_DIR/$VM_NAME"
++        mkdir $CIV_WORK_DIR/$VM_NAME
++}
++
++
++function copy_files_for_vm() {
++    echo "Copying file: $CIV_WORK_DIR/$VM_NAME"
++    mkdir $CIV_WORK_DIR/$VM_NAME/scripts
++    cp -r $CIV_WORK_DIR/scripts/aaf $CIV_WORK_DIR/$VM_NAME/scripts/
++    cp OVMF.fd $CIV_WORK_DIR/$VM_NAME/
++    ls -la $CIV_WORK_DIR/$VM_NAME/
++    cp $CIV_WORK_DIR/scripts/rpmb_dev $CIV_WORK_DIR/$VM_NAME/scripts/
++}
+ 
+ #-------------    main processes    -------------
+ 
+@@ -623,10 +666,12 @@ ubu_install_qemu_gvt
+ ubu_build_ovmf_gvt
+ ubu_enable_host_gvt
+ ubu_update_fw
+-
+ install_vm_manager
++
++create_vm_dir
+ create_aaf_dir
+-setup_civ_ini
++setup_civ_ini $VSOCK_ID $ADB_PORT $VM_NAME
++copy_files_for_vm
+ 
+ prepare_required_scripts
+ ubu_install_swtpm
+diff --git a/scripts/setup_host_multiple_android_vm.sh b/scripts/setup_host_multiple_android_vm.sh
+new file mode 100644
+index 0000000..c94b4aa
+--- /dev/null
++++ b/scripts/setup_host_multiple_android_vm.sh
+@@ -0,0 +1,101 @@
++#!/bin/bash
++
++# Copyright (c) 2023 Intel Corporation.
++# All rights reserved.
++#
++# SPDX-License-Identifier: Apache-2.0
++
++set -eE
++
++#---------      Global variable     -------------------
++CIV_DIR=$(pwd)
++USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
++VM_NAME=Android-CIV
++VSOCK_ID=3
++NUMBER_OF_VM=1
++ADB_PORT=5555
++INITIAL_VM_NUM=1
++SETUP_HOST_DONE=false
++
++function setup_files() {
++    echo "Creating setup for VMs"
++    if [ -e $USER_HOME/.intel/.civ ]
++    then
++        SETUP_HOST_DONE=true
++    fi
++    new_dir=$CIV_DIR/$VM_NAME
++    if [ -n "$(ls -d ${new_dir}?)" ]; then
++	dir_arr=$(ls -d ${new_dir}?)
++	directories=( $dir_arr )
++	last_vm=${directories[-1]}
++	final_vm_name="${last_vm##*/}"
++	last_character=${final_vm_name: -1}
++	last_character=$((last_character+1))
++	ADB_PORT=$((5555+(last_character*2)))
++	VSOCK_ID=$((3+last_character))
++	INITIAL_VM_NUM=$last_character
++    fi
++
++    for (( i=$INITIAL_VM_NUM; i<$NUMBER_OF_VM+$INITIAL_VM_NUM; i++ ))
++    do
++	echo "Folder name: $CIV_DIR/${VM_NAME}${i}"
++
++	if [ $SETUP_HOST_DONE = false ]; then
++            ./scripts/setup_host.sh -v 3 -b $ADB_PORT -n ${VM_NAME}${i}
++	    SETUP_HOST_DONE=y
++	else
++	    ADB_PORT=$((ADB_PORT+(i*2)))
++	    ./scripts/create_vms.sh -v $((i+3)) -b $ADB_PORT -n ${VM_NAME}${i}
++	fi
++    done
++
++    echo "Please reboot"
++}
++
++function copy_files_for_vm() {
++    echo "Copying to folder $VM_NAME$1"
++    cp $CIV_DIR/$VM_NAME$INITIAL_VM_NUM/android.qcow2 $CIV_DIR/$VM_NAME$1
++    cp -r $CIV_DIR/$VM_NAME$INITIAL_VM_NUM/vtpm0 $CIV_DIR/$VM_NAME$1
```

Comment:
For the two copy operations, check that files exists otherwise display error message and exit